### PR TITLE
Tips for rpc perf

### PR DESCRIPTION
**Pull request by m-shaka**
Added tips to improve IDE performance when using RPC

### CODE REVIEW

**Comment by yusukebe** [comment]
Hey @m-shaka 

Good descriptions! These are needed. I've left a comment. Check it.

**Comment by yusukebe** [comment]
I'm not sure this will help you, but I've created the repo to use `d.ts`: https://github.com/yusukebe/hono-rpc-with-d-ts

**Comment by m-shaka** [comment]
Great!
I was thinking of separating the server project and the client one using a monorepo tool as I'm doing in my job, but your working example looks perfect as a single repo.
I'll add a description about monorepo and the link to your repository to the doc.

File: docs/guides/rpc.md [hunk 1]
```diff
@@ -395,3 +395,91 @@ export type AppType = typeof routes
...
+    BlankInput,
+    BlankEnv
+  >('foo/:id', (c) => c.json({ ok: true }, 200))
+```
+
+This is a type instantiation for a single route. While the user doesn't need to write these type arguments manually, which is a good thing, it's known that type instantiation takes much time. `tsserver` does this time consuming task every time you use the app. If you have a lot of routes, this can slow down your IDE significantly.
+
+However, we have some tips to mitigate this issue.
+
+#### compile your app before using it (recommended)
```
[the remark this answers is not in the document]
Comment (m-shaka) [reply docs/guides/rpc.md:430]: Updated!

**Comment by yusukebe** [comment]
@m-shaka 

Hey. This is another approach in which @nakasyou tried similar things. 

https://github.com/yusukebe/hono-rpc-with-d-ts/tree/client-types

In this branch, `client/client.ts` uses [the actual client types](https://github.com/yusukebe/hono-rpc-with-d-ts/blob/client-types/types/client-types/client.d.ts) emitted from `client-types/client.ts`. This may be the fastest. This is interesting.

**Comment by yusukebe** [comment]
Oooops. Please ignore the CI is failing.

**Comment by m-shaka** [comment]
Oh, I see.
Or rather it's better to export a client from the server, right?

```ts
// server/client.ts
import routes from './server'
import { hc } from 'hono/client'

type Routes = typeof routes
export const getClient: typeof hc<Routes> = (...args) => hc<Routes>(...args)

// client/client.ts
import { getClient } from '../dist/client'
const client = getClient('http://localhost:3000')

const res = await client.index.$get()
const data = await res.json()
console.log(data)
```

**Comment by m-shaka** [comment]
Some tricks are needed to calculate types on compiling.

```typescript
// server/client.ts
import routes from './server'
import { hc } from 'hono/client'

type Routes = typeof routes

const client = hc<Routes>('http://localhost:3000')

export type Client = typeof client

export const getClient = (...args: Parameters<typeof hc<Routes>>): Client =>
    hc<Routes>(...args)
```

**Comment by yusukebe** [comment]
@m-shaka 

Ah, cool! It's nitpicky, but how about renaming `getClient` to `hcWithType`?

**Comment by yusukebe** [comment]
@nakasyou any thoughts on it?

**Comment by nakasyou** [comment]
It looks cool! And I like `hcWithType` better than `getClient`.

**Comment by m-shaka** [comment]
Thank you, both of you! I revised the doc

**Comment by yusukebe** [comment]
Hi @m-shaka 

I've updated the format of the docs and added a super short description for the `server`. Let's go with it. Thanks!

**Pull request merged**